analyzers/src/SonarAnalyzer.Common/Helpers/SyntaxNodeExtensions.cs

```diff
@@ -53,6 +53,9 @@ public static string GetMappedFilePathFromRoot(this SyntaxNode root)
         return root.SyntaxTree.FilePath;
     }
 
-    internal static TSyntaxKind Kind<TSyntaxKind>(this SyntaxNode node) where TSyntaxKind : struct, Enum => // internal to not be confused with e.g. CSharp.SyntaxNode.Kind()
-        node == null ? default : (TSyntaxKind)Enum.ToObject(typeof(TSyntaxKind), node.RawKind);
+    public static TSyntaxKind Kind<TSyntaxKind>(this SyntaxNode node) where TSyntaxKind : struct, Enum => // internal to not be confused with e.g. CSharp.SyntaxNode.Kind()
```

Comment:
Either keep the internal or remove the comment
```suggestion
    internal static TSyntaxKind Kind<TSyntaxKind>(this SyntaxNode node) where TSyntaxKind : struct, Enum => // internal to not be confused with e.g. CSharp.SyntaxNode.Kind()
```